core/embed/rust/src/translations/translated_string.rs

```diff
@@ -4,16 +4,15 @@ use super::blob::Translations;
 pub use super::generated::translated_string::TranslatedString;
 
 impl TranslatedString {
-    pub(super) fn translate<'a>(self, source: Option<&'a Translations>) -> &'a str {
+    pub fn translate<'a>(self, source: Option<&'a Translations>) -> &'a str {
         source
             .and_then(|s| s.translation(self as _))
             .unwrap_or(self.untranslated())
     }
 
-    pub fn map_translated<F, T>(self, fun: F) -> T
+    pub fn map_translated<'a, F, T>(self, fun: F) -> T
     where
-        F: for<'a> FnOnce(&'a str) -> T,
-        T: 'static,
+        F: FnOnce(&'a str) -> T,
```

Comment:
Come on man!
I went to all the trouble to forbid the string from escaping the closure, and yet that's exactly what you are reintroducing. `x.as_ref()` is no more safe than `x.map(|c| c)`. The lifetime of the translated string is _independent of the lifetime of its TString_ -- namely, it can become invalid any time it pleases. Hence the restrictive lifetimes.

Reply:
sorry, i misunderstood the whole idea. 
hopefully this will be better: https://github.com/trezor/trezor-firmware/pull/3649/commits/a045dce2ae1f54bd6d8eb5fccee051fa194de545